buildflow/runtime/ray_io/file_io.py

```diff
@@ -52,9 +56,22 @@ async def _write(self, elements: Union[ray.data.Dataset,
         if isinstance(elements, ray.data.Dataset):
             if self._format == FileFormat.PARQUET:
                 elements.write_parquet(self._path)
+            elif self._format == FileFormat.CSV:
+                elements.write_csv(self._path)
+            elif self._format == FileFormat.JSON:
+                elements.write_json(self._path)
         else:
             if self._format == FileFormat.PARQUET:
                 exists = os.path.exists(self._path)
                 fastparquet.write(self._path,
                                   pd.DataFrame.from_records(elements),
                                   append=exists)
+            elif self._format == FileFormat.CSV:
+                with open(self._path, 'a', newline='') as output_file:
+                    keys = elements[0].keys()
```

Comment:
might be worth adding a check to make sure there are elements in the list. Probably safe to just skip writing if there are no elements